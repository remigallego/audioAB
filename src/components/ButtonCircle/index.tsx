import React from "react";
import styled from "styled-components";

interface Props {
  isA: boolean;
  toggleButton: () => void;
}

const ButtonCircle = (props: Props) => {
  return (
    <Circle onClick={props.toggleButton}>
      <TextWrapper>
        <PrimaryText>{props.isA ? "A" : "B"}</PrimaryText>
        <SecondaryText>{props.isA ? "B" : "A"}</SecondaryText>
      </TextWrapper>
    </Circle>
  );
};

export default ButtonCircle;

const Circle = styled.section`
  width: 150px;
  height: 150px;
  background-color: blue;
  font-family: "DM Serif Display";
  border-radius: 50%;
  position: relative;
  user-select: none;
`;

const PrimaryText = styled.section`
  color: white;
  font-size: 77px;
`;
const SecondaryText = styled.section`
  color: grey;
  font-size: 77px;
`;

const TextWrapper = styled.section`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-35%, -50%);

  display: flex;
  width: 100%;
`;
