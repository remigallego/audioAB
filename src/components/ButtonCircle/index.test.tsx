import React from "react";
import ButtonCircle from ".";
import renderer from "react-test-renderer";

test("ButtonCircle renders correctly", () => {
  const component = renderer.create(
    <ButtonCircle isA={true} toggleButton={() => null} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  /*   // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot(); */
});
