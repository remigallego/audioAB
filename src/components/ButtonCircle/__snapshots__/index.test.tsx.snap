// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCircle renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
      "borderRadius": "50%",
      "height": 100,
      "width": 100,
    }
  }
/>
`;
